dao/project.go

```diff
@@ -228,6 +192,38 @@ func GetUserRelevantProjects(userID int) ([]models.Project, error) {
 	return projects, nil
 }
 
+// GetUserRelevantProjects returns the projects based on publicity and user, disregarding the names etc.
+func GetUserRelevantProjects(query models.Project) ([]models.Project, error) {
```

Comment:
The function should be split into two func: GetUserRelevantProjects and GetPublicProjects.